internal/access/request.go

```diff
@@ -0,0 +1,26 @@
+package access
+
+import (
+	"net/http"
+
+	"gorm.io/gorm"
+
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+const RequestContextKey = "requestContext"
+
+// RequestContext stores the http.Request, and values derived from the request
+// like the authenticated user. It also provides a database transaction.
+type RequestContext struct {
+	Request       *http.Request
+	DBTxn         *gorm.DB
+	Authenticated Authenticated
+}
+
+// Authenticated stores data about the authenticated user. If the AccessKey or
+// User are nil, it indicates that no user was authenticated.
+type Authenticated struct {
+	AccessKey *models.AccessKey
```

Comment:
how do we know AccessKey is always going to be set? are these better off as functions?

Reply:
It won't always be set because `RequestContext` is used by unauthenticated endpoints as well. Right now it's doing a simple `!= nil` check, but in the future I think exposing more of this through methods would be good.